core/src/main/java/org/ehcache/util/ResourcePoolMerger.java

```diff
@@ -40,6 +41,9 @@ public ResourcePools validateAndMerge(ResourcePools existing, ResourcePools toBe
           existing.getPoolForResource(currentResourceType).isPersistent()) {
         throw new IllegalArgumentException("Persistence configuration cannot be updated");
       }
+      if(toBeUpdated.getPoolForResource(currentResourceType).getUnit() instanceof MemoryUnit) {
```

Comment:
While this test makes sense as long as #40 is not implemented, I believe we can have one expressed in the same way the persistent check is made - disallow changing the resource unit altogether.